akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,23 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case ActorSelectionMessage(msg, elements, _) ⇒ {
+      elements.foreach {
```

Comment:
I somehow cannot figure out why `foreach` on the elements makes sense here—each element descends one level into the tree …